src/Layouts/Partials/FillItem.vala

```diff
@@ -217,7 +217,73 @@ public class Akira.Layouts.Partials.FillItem : Gtk.Grid {
 
         color_picker = new Gtk.Grid ();
         color_picker.get_style_context ().add_class ("color-picker");
+        color_picker.row_spacing = 12;
+
+        // Document Grid - The Grid that will have all of the document saved colors
+        var document_grid = new Gtk.Grid ();
+        document_grid.margin_start = document_grid.margin_end = 40;
+        document_grid.hexpand = true;
+        var document_lbl = new Gtk.Label ("Document Colors");
+        document_lbl.halign = Gtk.Align.START;
+        document_grid.attach (document_lbl, 0, 0, 1, 1);
+        document_grid.row_spacing = 12;
+        var document_colors_grid = new Gtk.FlowBox ();
+        document_colors_grid.set_selection_mode (Gtk.SelectionMode.NONE);
+        document_colors_grid.column_spacing = 5;
+        document_colors_grid.row_spacing = 5;
+        document_grid.attach (document_colors_grid, 0, 1, 1, 1);
+
+        // Global Grid - The Grid that will have all of the document saved colors
+        var global_grid = new Gtk.Grid ();
+        global_grid.margin_start = global_grid.margin_end = 40;
+        global_grid.hexpand = true;
+        var global_lbl = new Gtk.Label ("Global Colors");
+        global_lbl.halign = Gtk.Align.START;
+        global_grid.attach (global_lbl, 0, 0, 1, 1);
+        global_grid.row_spacing = 12;
+        var global_colors_grid = new Gtk.FlowBox ();
+        global_colors_grid.max_children_per_line = 11;
+        global_colors_grid.set_selection_mode (Gtk.SelectionMode.NONE);
+        global_colors_grid.hexpand = true;
+        global_colors_grid.column_spacing = 5;
+        global_colors_grid.row_spacing = 5;
+
+        foreach (string global_color in settings.global_colors) {
+            var color_item = new Akira.Partials.RoundedColorButton (global_color);
+            color_item.clicked.connect (()=>{
+                var rgba_color = Gdk.RGBA ();
+                rgba_color.parse (color_item.background_color);
+                color_chooser_widget.set_rgba (rgba_color);
+            });
+            global_colors_grid.insert (color_item, -1);
```

Comment:
As we discussed on Matrix, this should be added into the last position before the add_button.
A sort function needs to be defined in order to allow it.